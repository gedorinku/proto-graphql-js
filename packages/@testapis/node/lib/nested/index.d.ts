import * as $protobuf from "protobufjs";
/** Namespace testapi. */
export namespace testapi {

    /** Namespace nested. */
    namespace nested {

        /** Properties of a ParentMessage. */
        interface IParentMessage {

            /** ParentMessage body */
            body?: (string|null);

            /** ParentMessage nested */
            nested?: (testapi.nested.ParentMessage.INestedMessage|null);

            /** ParentMessage nestedEnum */
            nestedEnum?: (testapi.nested.ParentMessage.NestedEnum|null);
        }

        /** Represents a ParentMessage. */
        class ParentMessage implements IParentMessage {

            /**
             * Constructs a new ParentMessage.
             * @param [properties] Properties to set
             */
            constructor(properties?: testapi.nested.IParentMessage);

            /** ParentMessage body. */
            public body: string;

            /** ParentMessage nested. */
            public nested?: (testapi.nested.ParentMessage.INestedMessage|null);

            /** ParentMessage nestedEnum. */
            public nestedEnum: testapi.nested.ParentMessage.NestedEnum;

            /**
             * Creates a new ParentMessage instance using the specified properties.
             * @param [properties] Properties to set
             * @returns ParentMessage instance
             */
            public static create(properties?: testapi.nested.IParentMessage): testapi.nested.ParentMessage;

            /**
             * Encodes the specified ParentMessage message. Does not implicitly {@link testapi.nested.ParentMessage.verify|verify} messages.
             * @param message ParentMessage message or plain object to encode
             * @param [writer] Writer to encode to
             * @returns Writer
             */
            public static encode(message: testapi.nested.IParentMessage, writer?: $protobuf.Writer): $protobuf.Writer;

            /**
             * Encodes the specified ParentMessage message, length delimited. Does not implicitly {@link testapi.nested.ParentMessage.verify|verify} messages.
             * @param message ParentMessage message or plain object to encode
             * @param [writer] Writer to encode to
             * @returns Writer
             */
            public static encodeDelimited(message: testapi.nested.IParentMessage, writer?: $protobuf.Writer): $protobuf.Writer;

            /**
             * Decodes a ParentMessage message from the specified reader or buffer.
             * @param reader Reader or buffer to decode from
             * @param [length] Message length if known beforehand
             * @returns ParentMessage
             * @throws {Error} If the payload is not a reader or valid buffer
             * @throws {$protobuf.util.ProtocolError} If required fields are missing
             */
            public static decode(reader: ($protobuf.Reader|Uint8Array), length?: number): testapi.nested.ParentMessage;

            /**
             * Decodes a ParentMessage message from the specified reader or buffer, length delimited.
             * @param reader Reader or buffer to decode from
             * @returns ParentMessage
             * @throws {Error} If the payload is not a reader or valid buffer
             * @throws {$protobuf.util.ProtocolError} If required fields are missing
             */
            public static decodeDelimited(reader: ($protobuf.Reader|Uint8Array)): testapi.nested.ParentMessage;

            /**
             * Verifies a ParentMessage message.
             * @param message Plain object to verify
             * @returns `null` if valid, otherwise the reason why it is not
             */
            public static verify(message: { [k: string]: any }): (string|null);

            /**
             * Creates a ParentMessage message from a plain object. Also converts values to their respective internal types.
             * @param object Plain object
             * @returns ParentMessage
             */
            public static fromObject(object: { [k: string]: any }): testapi.nested.ParentMessage;

            /**
             * Creates a plain object from a ParentMessage message. Also converts values to other types if specified.
             * @param message ParentMessage
             * @param [options] Conversion options
             * @returns Plain object
             */
            public static toObject(message: testapi.nested.ParentMessage, options?: $protobuf.IConversionOptions): { [k: string]: any };

            /**
             * Converts this ParentMessage to JSON.
             * @returns JSON object
             */
            public toJSON(): { [k: string]: any };
        }

        namespace ParentMessage {

            /** Properties of a NestedMessage. */
            interface INestedMessage {

                /** NestedMessage nestedBody */
                nestedBody?: (string|null);
            }

            /** Represents a NestedMessage. */
            class NestedMessage implements INestedMessage {

                /**
                 * Constructs a new NestedMessage.
                 * @param [properties] Properties to set
                 */
                constructor(properties?: testapi.nested.ParentMessage.INestedMessage);

                /** NestedMessage nestedBody. */
                public nestedBody: string;

                /**
                 * Creates a new NestedMessage instance using the specified properties.
                 * @param [properties] Properties to set
                 * @returns NestedMessage instance
                 */
                public static create(properties?: testapi.nested.ParentMessage.INestedMessage): testapi.nested.ParentMessage.NestedMessage;

                /**
                 * Encodes the specified NestedMessage message. Does not implicitly {@link testapi.nested.ParentMessage.NestedMessage.verify|verify} messages.
                 * @param message NestedMessage message or plain object to encode
                 * @param [writer] Writer to encode to
                 * @returns Writer
                 */
                public static encode(message: testapi.nested.ParentMessage.INestedMessage, writer?: $protobuf.Writer): $protobuf.Writer;

                /**
                 * Encodes the specified NestedMessage message, length delimited. Does not implicitly {@link testapi.nested.ParentMessage.NestedMessage.verify|verify} messages.
                 * @param message NestedMessage message or plain object to encode
                 * @param [writer] Writer to encode to
                 * @returns Writer
                 */
                public static encodeDelimited(message: testapi.nested.ParentMessage.INestedMessage, writer?: $protobuf.Writer): $protobuf.Writer;

                /**
                 * Decodes a NestedMessage message from the specified reader or buffer.
                 * @param reader Reader or buffer to decode from
                 * @param [length] Message length if known beforehand
                 * @returns NestedMessage
                 * @throws {Error} If the payload is not a reader or valid buffer
                 * @throws {$protobuf.util.ProtocolError} If required fields are missing
                 */
                public static decode(reader: ($protobuf.Reader|Uint8Array), length?: number): testapi.nested.ParentMessage.NestedMessage;

                /**
                 * Decodes a NestedMessage message from the specified reader or buffer, length delimited.
                 * @param reader Reader or buffer to decode from
                 * @returns NestedMessage
                 * @throws {Error} If the payload is not a reader or valid buffer
                 * @throws {$protobuf.util.ProtocolError} If required fields are missing
                 */
                public static decodeDelimited(reader: ($protobuf.Reader|Uint8Array)): testapi.nested.ParentMessage.NestedMessage;

                /**
                 * Verifies a NestedMessage message.
                 * @param message Plain object to verify
                 * @returns `null` if valid, otherwise the reason why it is not
                 */
                public static verify(message: { [k: string]: any }): (string|null);

                /**
                 * Creates a NestedMessage message from a plain object. Also converts values to their respective internal types.
                 * @param object Plain object
                 * @returns NestedMessage
                 */
                public static fromObject(object: { [k: string]: any }): testapi.nested.ParentMessage.NestedMessage;

                /**
                 * Creates a plain object from a NestedMessage message. Also converts values to other types if specified.
                 * @param message NestedMessage
                 * @param [options] Conversion options
                 * @returns Plain object
                 */
                public static toObject(message: testapi.nested.ParentMessage.NestedMessage, options?: $protobuf.IConversionOptions): { [k: string]: any };

                /**
                 * Converts this NestedMessage to JSON.
                 * @returns JSON object
                 */
                public toJSON(): { [k: string]: any };
            }

            /** NestedEnum enum. */
            enum NestedEnum {
                NESTED_ENUM_UNSPECIFIED = 0,
                FOO = 1,
                BAR = 2
            }
        }
    }
}
