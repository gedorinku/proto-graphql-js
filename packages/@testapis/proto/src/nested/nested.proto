syntax = "proto3";

package testapi.nested;

message ParentMessage {
  message NestedMessage {
    string nested_body = 1;
  }

  string body = 1;
  NestedMessage nested = 2;
}
