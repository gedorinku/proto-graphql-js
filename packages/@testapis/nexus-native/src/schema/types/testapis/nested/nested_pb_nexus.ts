// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: testapis/nested/nested.proto

import { objectType, inputObjectType, enumType, nullable, nonNull } from "nexus";
import * as $$testapis$node_native$lib$testapis$nested$nested_pb from "@testapis/node-native/lib/testapis/nested/nested_pb";
export type $$testapis$node_native$lib$testapis$nested$nested_pb$ParentMessage = $$testapis$node_native$lib$testapis$nested$nested_pb.ParentMessage;
export type $$testapis$node_native$lib$testapis$nested$nested_pb$ParentMessageNestedMessage = $$testapis$node_native$lib$testapis$nested$nested_pb.ParentMessage.NestedMessage;
export const ParentMessage = objectType({
    name: "ParentMessage",
    description: "",
    definition(t) {
        t.field("body", {
            type: nonNull("String"),
            description: "",
            resolve(root) { return root.getBody(); }
        });
        t.field("nested", {
            type: nullable("ParentMessageNestedMessage"),
            description: "",
            resolve(root) { return root.getNested() ?? null; }
        });
        t.field("nestedEnum", {
            type: nullable("ParentMessageNestedEnum"),
            description: "",
            resolve(root) { return root.getNestedEnum(); }
        });
    },
    sourceType: { module: __filename, export: "$$testapis$node_native$lib$testapis$nested$nested_pb$ParentMessage" }
});
export const ParentMessageNestedMessage = objectType({
    name: "ParentMessageNestedMessage",
    description: "",
    definition(t) {
        t.field("nestedBody", {
            type: nonNull("String"),
            description: "",
            resolve(root) { return root.getNestedBody(); }
        });
    },
    sourceType: { module: __filename, export: "$$testapis$node_native$lib$testapis$nested$nested_pb$ParentMessageNestedMessage" }
});
export const ParentMessageInput = inputObjectType({
    name: "ParentMessageInput",
    description: "",
    definition(t) {
        t.field("body", {
            type: nonNull("String"),
            description: ""
        });
        t.field("nested", {
            type: nullable("ParentMessageNestedMessageInput"),
            description: ""
        });
        t.field("nestedEnum", {
            type: nullable("ParentMessageNestedEnum"),
            description: ""
        });
    }
});
export const ParentMessageNestedMessageInput = inputObjectType({
    name: "ParentMessageNestedMessageInput",
    description: "",
    definition(t) {
        t.field("nestedBody", {
            type: nonNull("String"),
            description: ""
        });
    }
});
export const ParentMessageNestedEnum = enumType({
    name: "ParentMessageNestedEnum",
    description: "",
    members: [
        {
            name: "NESTED_ENUM_UNSPECIFIED",
            value: 0
        },
        {
            name: "FOO",
            value: 1
        },
        {
            name: "BAR",
            value: 2
        }
    ]
});
