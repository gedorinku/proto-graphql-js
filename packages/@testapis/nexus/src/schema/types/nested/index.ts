export * from "./nested_nexus_pb";
