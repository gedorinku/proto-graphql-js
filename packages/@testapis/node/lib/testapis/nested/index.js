/*eslint-disable block-scoped-var, id-length, no-control-regex, no-magic-numbers, no-prototype-builtins, no-redeclare, no-shadow, no-var, sort-vars*/
"use strict";

var $protobuf = require("protobufjs/minimal");

// Common aliases
var $Reader = $protobuf.Reader, $Writer = $protobuf.Writer, $util = $protobuf.util;

// Exported root namespace
var $root = $protobuf.roots["default"] || ($protobuf.roots["default"] = {});

$root.testapis = (function() {

    /**
     * Namespace testapis.
     * @exports testapis
     * @namespace
     */
    var testapis = {};

    testapis.nested = (function() {

        /**
         * Namespace nested.
         * @memberof testapis
         * @namespace
         */
        var nested = {};

        nested.ParentMessage = (function() {

            /**
             * Properties of a ParentMessage.
             * @memberof testapis.nested
             * @interface IParentMessage
             * @property {string|null} [body] ParentMessage body
             * @property {testapis.nested.ParentMessage.INestedMessage|null} [nested] ParentMessage nested
             * @property {testapis.nested.ParentMessage.NestedEnum|null} [nestedEnum] ParentMessage nestedEnum
             */

            /**
             * Constructs a new ParentMessage.
             * @memberof testapis.nested
             * @classdesc Represents a ParentMessage.
             * @implements IParentMessage
             * @constructor
             * @param {testapis.nested.IParentMessage=} [properties] Properties to set
             */
            function ParentMessage(properties) {
                if (properties)
                    for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                        if (properties[keys[i]] != null)
                            this[keys[i]] = properties[keys[i]];
            }

            /**
             * ParentMessage body.
             * @member {string} body
             * @memberof testapis.nested.ParentMessage
             * @instance
             */
            ParentMessage.prototype.body = "";

            /**
             * ParentMessage nested.
             * @member {testapis.nested.ParentMessage.INestedMessage|null|undefined} nested
             * @memberof testapis.nested.ParentMessage
             * @instance
             */
            ParentMessage.prototype.nested = null;

            /**
             * ParentMessage nestedEnum.
             * @member {testapis.nested.ParentMessage.NestedEnum} nestedEnum
             * @memberof testapis.nested.ParentMessage
             * @instance
             */
            ParentMessage.prototype.nestedEnum = 0;

            /**
             * Creates a new ParentMessage instance using the specified properties.
             * @function create
             * @memberof testapis.nested.ParentMessage
             * @static
             * @param {testapis.nested.IParentMessage=} [properties] Properties to set
             * @returns {testapis.nested.ParentMessage} ParentMessage instance
             */
            ParentMessage.create = function create(properties) {
                return new ParentMessage(properties);
            };

            /**
             * Encodes the specified ParentMessage message. Does not implicitly {@link testapis.nested.ParentMessage.verify|verify} messages.
             * @function encode
             * @memberof testapis.nested.ParentMessage
             * @static
             * @param {testapis.nested.IParentMessage} message ParentMessage message or plain object to encode
             * @param {$protobuf.Writer} [writer] Writer to encode to
             * @returns {$protobuf.Writer} Writer
             */
            ParentMessage.encode = function encode(message, writer) {
                if (!writer)
                    writer = $Writer.create();
                if (message.body != null && Object.hasOwnProperty.call(message, "body"))
                    writer.uint32(/* id 1, wireType 2 =*/10).string(message.body);
                if (message.nested != null && Object.hasOwnProperty.call(message, "nested"))
                    $root.testapis.nested.ParentMessage.NestedMessage.encode(message.nested, writer.uint32(/* id 2, wireType 2 =*/18).fork()).ldelim();
                if (message.nestedEnum != null && Object.hasOwnProperty.call(message, "nestedEnum"))
                    writer.uint32(/* id 3, wireType 0 =*/24).int32(message.nestedEnum);
                return writer;
            };

            /**
             * Encodes the specified ParentMessage message, length delimited. Does not implicitly {@link testapis.nested.ParentMessage.verify|verify} messages.
             * @function encodeDelimited
             * @memberof testapis.nested.ParentMessage
             * @static
             * @param {testapis.nested.IParentMessage} message ParentMessage message or plain object to encode
             * @param {$protobuf.Writer} [writer] Writer to encode to
             * @returns {$protobuf.Writer} Writer
             */
            ParentMessage.encodeDelimited = function encodeDelimited(message, writer) {
                return this.encode(message, writer).ldelim();
            };

            /**
             * Decodes a ParentMessage message from the specified reader or buffer.
             * @function decode
             * @memberof testapis.nested.ParentMessage
             * @static
             * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
             * @param {number} [length] Message length if known beforehand
             * @returns {testapis.nested.ParentMessage} ParentMessage
             * @throws {Error} If the payload is not a reader or valid buffer
             * @throws {$protobuf.util.ProtocolError} If required fields are missing
             */
            ParentMessage.decode = function decode(reader, length) {
                if (!(reader instanceof $Reader))
                    reader = $Reader.create(reader);
                var end = length === undefined ? reader.len : reader.pos + length, message = new $root.testapis.nested.ParentMessage();
                while (reader.pos < end) {
                    var tag = reader.uint32();
                    switch (tag >>> 3) {
                    case 1:
                        message.body = reader.string();
                        break;
                    case 2:
                        message.nested = $root.testapis.nested.ParentMessage.NestedMessage.decode(reader, reader.uint32());
                        break;
                    case 3:
                        message.nestedEnum = reader.int32();
                        break;
                    default:
                        reader.skipType(tag & 7);
                        break;
                    }
                }
                return message;
            };

            /**
             * Decodes a ParentMessage message from the specified reader or buffer, length delimited.
             * @function decodeDelimited
             * @memberof testapis.nested.ParentMessage
             * @static
             * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
             * @returns {testapis.nested.ParentMessage} ParentMessage
             * @throws {Error} If the payload is not a reader or valid buffer
             * @throws {$protobuf.util.ProtocolError} If required fields are missing
             */
            ParentMessage.decodeDelimited = function decodeDelimited(reader) {
                if (!(reader instanceof $Reader))
                    reader = new $Reader(reader);
                return this.decode(reader, reader.uint32());
            };

            /**
             * Verifies a ParentMessage message.
             * @function verify
             * @memberof testapis.nested.ParentMessage
             * @static
             * @param {Object.<string,*>} message Plain object to verify
             * @returns {string|null} `null` if valid, otherwise the reason why it is not
             */
            ParentMessage.verify = function verify(message) {
                if (typeof message !== "object" || message === null)
                    return "object expected";
                if (message.body != null && message.hasOwnProperty("body"))
                    if (!$util.isString(message.body))
                        return "body: string expected";
                if (message.nested != null && message.hasOwnProperty("nested")) {
                    var error = $root.testapis.nested.ParentMessage.NestedMessage.verify(message.nested);
                    if (error)
                        return "nested." + error;
                }
                if (message.nestedEnum != null && message.hasOwnProperty("nestedEnum"))
                    switch (message.nestedEnum) {
                    default:
                        return "nestedEnum: enum value expected";
                    case 0:
                    case 1:
                    case 2:
                        break;
                    }
                return null;
            };

            /**
             * Creates a ParentMessage message from a plain object. Also converts values to their respective internal types.
             * @function fromObject
             * @memberof testapis.nested.ParentMessage
             * @static
             * @param {Object.<string,*>} object Plain object
             * @returns {testapis.nested.ParentMessage} ParentMessage
             */
            ParentMessage.fromObject = function fromObject(object) {
                if (object instanceof $root.testapis.nested.ParentMessage)
                    return object;
                var message = new $root.testapis.nested.ParentMessage();
                if (object.body != null)
                    message.body = String(object.body);
                if (object.nested != null) {
                    if (typeof object.nested !== "object")
                        throw TypeError(".testapis.nested.ParentMessage.nested: object expected");
                    message.nested = $root.testapis.nested.ParentMessage.NestedMessage.fromObject(object.nested);
                }
                switch (object.nestedEnum) {
                case "NESTED_ENUM_UNSPECIFIED":
                case 0:
                    message.nestedEnum = 0;
                    break;
                case "FOO":
                case 1:
                    message.nestedEnum = 1;
                    break;
                case "BAR":
                case 2:
                    message.nestedEnum = 2;
                    break;
                }
                return message;
            };

            /**
             * Creates a plain object from a ParentMessage message. Also converts values to other types if specified.
             * @function toObject
             * @memberof testapis.nested.ParentMessage
             * @static
             * @param {testapis.nested.ParentMessage} message ParentMessage
             * @param {$protobuf.IConversionOptions} [options] Conversion options
             * @returns {Object.<string,*>} Plain object
             */
            ParentMessage.toObject = function toObject(message, options) {
                if (!options)
                    options = {};
                var object = {};
                if (options.defaults) {
                    object.body = "";
                    object.nested = null;
                    object.nestedEnum = options.enums === String ? "NESTED_ENUM_UNSPECIFIED" : 0;
                }
                if (message.body != null && message.hasOwnProperty("body"))
                    object.body = message.body;
                if (message.nested != null && message.hasOwnProperty("nested"))
                    object.nested = $root.testapis.nested.ParentMessage.NestedMessage.toObject(message.nested, options);
                if (message.nestedEnum != null && message.hasOwnProperty("nestedEnum"))
                    object.nestedEnum = options.enums === String ? $root.testapis.nested.ParentMessage.NestedEnum[message.nestedEnum] : message.nestedEnum;
                return object;
            };

            /**
             * Converts this ParentMessage to JSON.
             * @function toJSON
             * @memberof testapis.nested.ParentMessage
             * @instance
             * @returns {Object.<string,*>} JSON object
             */
            ParentMessage.prototype.toJSON = function toJSON() {
                return this.constructor.toObject(this, $protobuf.util.toJSONOptions);
            };

            ParentMessage.NestedMessage = (function() {

                /**
                 * Properties of a NestedMessage.
                 * @memberof testapis.nested.ParentMessage
                 * @interface INestedMessage
                 * @property {string|null} [nestedBody] NestedMessage nestedBody
                 */

                /**
                 * Constructs a new NestedMessage.
                 * @memberof testapis.nested.ParentMessage
                 * @classdesc Represents a NestedMessage.
                 * @implements INestedMessage
                 * @constructor
                 * @param {testapis.nested.ParentMessage.INestedMessage=} [properties] Properties to set
                 */
                function NestedMessage(properties) {
                    if (properties)
                        for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                            if (properties[keys[i]] != null)
                                this[keys[i]] = properties[keys[i]];
                }

                /**
                 * NestedMessage nestedBody.
                 * @member {string} nestedBody
                 * @memberof testapis.nested.ParentMessage.NestedMessage
                 * @instance
                 */
                NestedMessage.prototype.nestedBody = "";

                /**
                 * Creates a new NestedMessage instance using the specified properties.
                 * @function create
                 * @memberof testapis.nested.ParentMessage.NestedMessage
                 * @static
                 * @param {testapis.nested.ParentMessage.INestedMessage=} [properties] Properties to set
                 * @returns {testapis.nested.ParentMessage.NestedMessage} NestedMessage instance
                 */
                NestedMessage.create = function create(properties) {
                    return new NestedMessage(properties);
                };

                /**
                 * Encodes the specified NestedMessage message. Does not implicitly {@link testapis.nested.ParentMessage.NestedMessage.verify|verify} messages.
                 * @function encode
                 * @memberof testapis.nested.ParentMessage.NestedMessage
                 * @static
                 * @param {testapis.nested.ParentMessage.INestedMessage} message NestedMessage message or plain object to encode
                 * @param {$protobuf.Writer} [writer] Writer to encode to
                 * @returns {$protobuf.Writer} Writer
                 */
                NestedMessage.encode = function encode(message, writer) {
                    if (!writer)
                        writer = $Writer.create();
                    if (message.nestedBody != null && Object.hasOwnProperty.call(message, "nestedBody"))
                        writer.uint32(/* id 1, wireType 2 =*/10).string(message.nestedBody);
                    return writer;
                };

                /**
                 * Encodes the specified NestedMessage message, length delimited. Does not implicitly {@link testapis.nested.ParentMessage.NestedMessage.verify|verify} messages.
                 * @function encodeDelimited
                 * @memberof testapis.nested.ParentMessage.NestedMessage
                 * @static
                 * @param {testapis.nested.ParentMessage.INestedMessage} message NestedMessage message or plain object to encode
                 * @param {$protobuf.Writer} [writer] Writer to encode to
                 * @returns {$protobuf.Writer} Writer
                 */
                NestedMessage.encodeDelimited = function encodeDelimited(message, writer) {
                    return this.encode(message, writer).ldelim();
                };

                /**
                 * Decodes a NestedMessage message from the specified reader or buffer.
                 * @function decode
                 * @memberof testapis.nested.ParentMessage.NestedMessage
                 * @static
                 * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
                 * @param {number} [length] Message length if known beforehand
                 * @returns {testapis.nested.ParentMessage.NestedMessage} NestedMessage
                 * @throws {Error} If the payload is not a reader or valid buffer
                 * @throws {$protobuf.util.ProtocolError} If required fields are missing
                 */
                NestedMessage.decode = function decode(reader, length) {
                    if (!(reader instanceof $Reader))
                        reader = $Reader.create(reader);
                    var end = length === undefined ? reader.len : reader.pos + length, message = new $root.testapis.nested.ParentMessage.NestedMessage();
                    while (reader.pos < end) {
                        var tag = reader.uint32();
                        switch (tag >>> 3) {
                        case 1:
                            message.nestedBody = reader.string();
                            break;
                        default:
                            reader.skipType(tag & 7);
                            break;
                        }
                    }
                    return message;
                };

                /**
                 * Decodes a NestedMessage message from the specified reader or buffer, length delimited.
                 * @function decodeDelimited
                 * @memberof testapis.nested.ParentMessage.NestedMessage
                 * @static
                 * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
                 * @returns {testapis.nested.ParentMessage.NestedMessage} NestedMessage
                 * @throws {Error} If the payload is not a reader or valid buffer
                 * @throws {$protobuf.util.ProtocolError} If required fields are missing
                 */
                NestedMessage.decodeDelimited = function decodeDelimited(reader) {
                    if (!(reader instanceof $Reader))
                        reader = new $Reader(reader);
                    return this.decode(reader, reader.uint32());
                };

                /**
                 * Verifies a NestedMessage message.
                 * @function verify
                 * @memberof testapis.nested.ParentMessage.NestedMessage
                 * @static
                 * @param {Object.<string,*>} message Plain object to verify
                 * @returns {string|null} `null` if valid, otherwise the reason why it is not
                 */
                NestedMessage.verify = function verify(message) {
                    if (typeof message !== "object" || message === null)
                        return "object expected";
                    if (message.nestedBody != null && message.hasOwnProperty("nestedBody"))
                        if (!$util.isString(message.nestedBody))
                            return "nestedBody: string expected";
                    return null;
                };

                /**
                 * Creates a NestedMessage message from a plain object. Also converts values to their respective internal types.
                 * @function fromObject
                 * @memberof testapis.nested.ParentMessage.NestedMessage
                 * @static
                 * @param {Object.<string,*>} object Plain object
                 * @returns {testapis.nested.ParentMessage.NestedMessage} NestedMessage
                 */
                NestedMessage.fromObject = function fromObject(object) {
                    if (object instanceof $root.testapis.nested.ParentMessage.NestedMessage)
                        return object;
                    var message = new $root.testapis.nested.ParentMessage.NestedMessage();
                    if (object.nestedBody != null)
                        message.nestedBody = String(object.nestedBody);
                    return message;
                };

                /**
                 * Creates a plain object from a NestedMessage message. Also converts values to other types if specified.
                 * @function toObject
                 * @memberof testapis.nested.ParentMessage.NestedMessage
                 * @static
                 * @param {testapis.nested.ParentMessage.NestedMessage} message NestedMessage
                 * @param {$protobuf.IConversionOptions} [options] Conversion options
                 * @returns {Object.<string,*>} Plain object
                 */
                NestedMessage.toObject = function toObject(message, options) {
                    if (!options)
                        options = {};
                    var object = {};
                    if (options.defaults)
                        object.nestedBody = "";
                    if (message.nestedBody != null && message.hasOwnProperty("nestedBody"))
                        object.nestedBody = message.nestedBody;
                    return object;
                };

                /**
                 * Converts this NestedMessage to JSON.
                 * @function toJSON
                 * @memberof testapis.nested.ParentMessage.NestedMessage
                 * @instance
                 * @returns {Object.<string,*>} JSON object
                 */
                NestedMessage.prototype.toJSON = function toJSON() {
                    return this.constructor.toObject(this, $protobuf.util.toJSONOptions);
                };

                return NestedMessage;
            })();

            /**
             * NestedEnum enum.
             * @name testapis.nested.ParentMessage.NestedEnum
             * @enum {number}
             * @property {number} NESTED_ENUM_UNSPECIFIED=0 NESTED_ENUM_UNSPECIFIED value
             * @property {number} FOO=1 FOO value
             * @property {number} BAR=2 BAR value
             */
            ParentMessage.NestedEnum = (function() {
                var valuesById = {}, values = Object.create(valuesById);
                values[valuesById[0] = "NESTED_ENUM_UNSPECIFIED"] = 0;
                values[valuesById[1] = "FOO"] = 1;
                values[valuesById[2] = "BAR"] = 2;
                return values;
            })();

            return ParentMessage;
        })();

        return nested;
    })();

    return testapis;
})();

module.exports = $root;
