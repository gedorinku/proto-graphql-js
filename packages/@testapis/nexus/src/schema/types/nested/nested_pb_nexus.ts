// Code generated by protoc-gen-nexus. DO NOT EDIT.
// source: nested/nested.proto

import { objectType, enumType } from "nexus";
import * as $$testapis$node$lib$nested$nested_pb from "@testapis/node/lib/nested/nested_pb";
export type $$testapis$node$lib$nested$nested_pb$ParentMessage = $$testapis$node$lib$nested$nested_pb.ParentMessage;
export type $$testapis$node$lib$nested$nested_pb$ParentMessageNestedMessage = $$testapis$node$lib$nested$nested_pb.ParentMessage.NestedMessage;
export const ParentMessage = objectType({
    name: "ParentMessage",
    description: "",
    definition(t) {
        t.nonNull.string("body", {
            description: "",
            resolve(root) { return root.getBody(); }
        });
        t.nullable.field("nested", {
            description: "",
            type: "ParentMessageNestedMessage",
            resolve(root) { return root.getNested() ?? null; }
        });
        t.nullable.field("nestedEnum", {
            description: "",
            type: "ParentMessageNestedEnum",
            resolve(root) { return root.getNestedEnum(); }
        });
    },
    sourceType: { module: __filename, export: "$$testapis$node$lib$nested$nested_pb$ParentMessage" }
});
export const ParentMessageNestedMessage = objectType({
    name: "ParentMessageNestedMessage",
    description: "",
    definition(t) {
        t.nonNull.string("nestedBody", {
            description: "",
            resolve(root) { return root.getNestedBody(); }
        });
    },
    sourceType: { module: __filename, export: "$$testapis$node$lib$nested$nested_pb$ParentMessageNestedMessage" }
});
export const ParentMessageNestedEnum = enumType({
    name: "ParentMessageNestedEnum",
    description: "",
    members: [
        {
            name: "NESTED_ENUM_UNSPECIFIED",
            value: 0
        },
        {
            name: "FOO",
            value: 1
        },
        {
            name: "BAR",
            value: 2
        }
    ]
});
