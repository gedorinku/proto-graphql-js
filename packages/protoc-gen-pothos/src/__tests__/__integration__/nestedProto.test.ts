import { testSchemaGeneration } from "../__helpers__/process.test.helper";

testSchemaGeneration("nested", "ts-proto", {
  schemaTests: [
    [
      "tests query",
      {
        extraSchema: `
          import { builder } from "./builder";
          import { ParentMessage$Ref } from "./testapis/nested/nested.pb.pothos";
          import { ParentMessage, ParentMessage_NestedEnum } from "@testapis/ts-proto/lib/testapis/nested/nested";

          builder.queryField("test", (t) => t.field({
            type: ParentMessage$Ref,
            resolve() {
              return ParentMessage.fromPartial({
                body: "hello",
                nested: {
                  nestedBody: "world",
                },
                nestedEnum: ParentMessage_NestedEnum.BAR
              });
            },
          }));
      `,
        testQuery: `
          query Test {
            test {
              body
              nested {
                nestedBody
              }
              nestedEnum
            }
          }
      `,
      },
    ],
  ],
});
