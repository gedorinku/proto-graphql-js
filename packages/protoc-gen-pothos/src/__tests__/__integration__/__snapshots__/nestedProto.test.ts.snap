// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema generation with ts-proto tests query 1`] = `
Object {
  "data": Object {
    "test": Object {
      "body": "hello",
      "nested": Object {
        "nestedBody": "world",
      },
      "nestedEnum": "BAR",
    },
  },
}
`;

exports[`schema generation with ts-proto tests query: schema.graphql 1`] = `
"type ParentMessage {
  body: String!
  nested: ParentMessageNestedMessage
  nestedEnum: ParentMessageNestedEnum
}

input ParentMessageInput {
  body: String!
  nested: ParentMessageNestedMessageInput
  nestedEnum: ParentMessageNestedEnum
}

enum ParentMessageNestedEnum {
  BAR
  FOO
}

type ParentMessageNestedMessage {
  nestedBody: String!
}

input ParentMessageNestedMessageInput {
  nestedBody: String!
}

type Query {
  test: ParentMessage!
}
"
`;
